governance/program/src/instruction.rs

```diff
@@ -0,0 +1,236 @@
+//! Program instructions
+
+use solana_program::{epoch_schedule::Slot, instruction::Instruction};
+
+use crate::state::Vote;
+
+/// Instructions supported by the Governance program
+#[derive(Clone)]
+#[allow(clippy::large_enum_variant)]
+pub enum GovernanceInstruction {
+    /// Creates Governance Realm account which aggregates governances for given Governance Mint and optional Council Mint.
+    ///
+    /// 1. `[writable]` Governance Realm account.
+    /// 2. `[]` Governance Token Mint.
+    /// 3. `[writable, signer]` Governances Token Holding account.
+    /// 4. `[signer]` Payer.
+    /// 5. `[]` System.
+    /// 6. `[]` SPL Token.
+    /// 7. `[]` Sysvar Rent.
+    /// 8. `[]` Council Token mint - optional.
+    /// 9. `[writable, signer]` Council Token Holding account - optional.
+    CreateGovernanceRealm {
+        /// UTF-8 encoded Governance Realm name.
+        name: String,
+    },
+
+    /// Deposits governing tokens (Governance or Council) to Governance Realm and establishes your voter weight to be used for voting within the Realm.
+    /// Note: If subsequent (top up) deposit is made and there are active votes for the Voter then the weights won't be updated automatically.
+    /// It can be done by relinquishing votes on active Proposals and voting again with the new weight.
+    ///
+    ///  0. `[]` Governance Realm account.
+    ///  1. `[]` Governing Token mint.
+    ///  2. `[writable]` Governing Token Holding account.
+    ///  3. `[writable]` Governing Token Source account.
```

Comment:
🤔 or we use token freezing instead like I mentioned in DM?  I'm liking the idea of using token freezing the more I think about it

Reply:
edit: freezing won't work!  So using a Token Delegate account to transfer tokens in to avoid Governance needing to deal with Token Multisigs seems like what we need here then

Reply:
I added `vote_authority` to split governing token ownership and the right to do actual voting. 
Also added a new instruction SetVoteAuthority to make it possible to change it. 
by default `vote_authority` is set to governing token owner.